### PR TITLE
Print unknown reason in SMT-LIB syntax

### PR DESCRIPTION
**Pull request by bclement-ocp**
The old format is preserved for legacy messages in the Alt-Ergo format, but the SMT-LIB format is the default one for pretty-printers, since we are moving towards preferring the SMT-LIB format.

Fixes #966

### CODE REVIEW

File: src/lib/reasoners/sat_solver_sig.ml [hunk 1]
```diff
@@ -44,14 +53,17 @@ type unknown_reason =
...
-let pp_unknown_reason_opt ppf = function
-  | None -> Fmt.pf ppf "Decided"
-  | Some ur -> pp_unknown_reason ppf ur
+  | Incomplete -> Fmt.pf ppf "incomplete"
+  | Memout -> Fmt.pf ppf "memout"
+  | Step_limit i -> Fmt.pf ppf "(:step-limit %i)" i
+  | Timeout t -> Fmt.pf ppf "(:timeout %a)" pp_timeout_reason t
+
+let pp_ae_unknown_reason_opt ppf = function
+  | None -> Fmt.pf ppf ":decided"
```
Comment (Stevendeo) [inline src/lib/reasoners/sat_solver_sig.ml]:
```suggestion
  | None -> Fmt.pf ppf "Decided"
```

File: src/lib/reasoners/sat_solver_sig.ml [hunk 2]
```diff
@@ -35,7 +35,16 @@ type timeout_reason =
   | Assume
   | ProofSearch
   | ModelGen
-[@@deriving show]
+
+let pp_timeout_reason ppf = function
```
Comment (Stevendeo) [inline src/lib/reasoners/sat_solver_sig.ml]:
```suggestion
let pp_smt_timeout_reason ppf = function
```

File: src/lib/reasoners/sat_solver_sig.ml (hunk 1 above)
Comment (bclement-ocp) [reply src/lib/reasoners/sat_solver_sig.ml]: Oops :grimacing:

**Comment by Stevendeo** [comment]
Closing this PR, which is now in https://github.com/OCamlPro/alt-ergo/pull/975

**Pull request closed without merging**